Tools/MonoGame.Tools.Tests/CharacterRegionTypeConverterTests.cs

```diff
@@ -0,0 +1,102 @@
+            Assert.AreEqual('A', result.Start);
+            Assert.AreEqual('A', result.End);
+        }
+
+        [Test]
+        public void ConvertFrom_AsciiRange_ReturnsCharacterRegion()
+        {
+            var result = (CharacterRegion)converter.ConvertFrom(null, null, "A-Z");
+            Assert.AreEqual('A', result.Start);
+            Assert.AreEqual('Z', result.End);
+        }
+
+        [Test]
+        public void ConvertFrom_SingleUnicodeCharacter_ReturnsSingleCharacterRegion()
+        {
+            var result = (CharacterRegion)converter.ConvertFrom(null, null, "あ");
+            Assert.AreEqual('あ', result.Start);
+            Assert.AreEqual('あ', result.End);
+        }
+
+        [Test]
+        public void ConvertFrom_UnicodeRange_ReturnsCharacterRegion()
+        {
+            var result = (CharacterRegion)converter.ConvertFrom(null, null, "あ-ん");
+            Assert.AreEqual('あ', result.Start);
+            Assert.AreEqual('ん', result.End);
+        }
+
+        [Test]
+        public void ConvertFrom_CyrillicRange_ReturnsCharacterRegion()
+        {
+            var result = (CharacterRegion)converter.ConvertFrom(null, null, "А-Я");
+            Assert.AreEqual('А', result.Start);
+            Assert.AreEqual('Я', result.End);
+        }
+
+        [Test]
+        public void ConvertFrom_HexadecimalCodePoint_ReturnsSingleCharacterRegion()
+        {
+            var result = (CharacterRegion)converter.ConvertFrom(null, null, "0x3042");
+            Assert.AreEqual('あ', result.Start);
+            Assert.AreEqual('あ', result.End);
+        }
+
+        [Test]
+        public void ConvertFrom_DecimalEntityCodePoint_ReturnsSingleCharacterRegion()
+        {
+            var result = (CharacterRegion)converter.ConvertFrom(null, null, "&#12354;");
+            Assert.AreEqual('あ', result.Start);
+            Assert.AreEqual('あ', result.End);
+        }
+
+        [Test]
+        public void ConvertFrom_IntegerCodePoint_ReturnsSingleCharacterRegion()
+        {
+            var result = (CharacterRegion)converter.ConvertFrom(null, null, "12354"); // 'あ' in decimal
+            Assert.AreEqual('あ', result.Start);
+            Assert.AreEqual('あ', result.End);
+        }
+
+        [Test]
+        public void ConvertFrom_InvalidInput_ThrowsArgumentException()
+        {
+            Assert.Throws<ArgumentException>(() => converter.ConvertFrom(null, null, ""));
+            Assert.Throws<ArgumentException>(() => converter.ConvertFrom(null, null, "A--Z"));
+            Assert.Throws<ArgumentException>(() => converter.ConvertFrom(null, null, "Invalid"));
+            Assert.Throws<ArgumentException>(() => converter.ConvertFrom(null, null, "0xZZ"));
+        }
+
+        [Test]
+        public void ConvertFrom_InvalidRangeOrder_ThrowsArgumentException()
+        {
+            Assert.Throws<ArgumentException>(() => converter.ConvertFrom(null, null, "Z-A"));
```

Comment:
@kjpou1 I believe this unit test and the next one should catch the start > end error, for both standard and multi-byte characters.